tests/reflect/pos/QuotRemInt.hs

```diff
@@ -0,0 +1,23 @@
+-- | Reflection of a function using `quot` and `rem` was not working,
+-- because they were not defined in the refinement logic until
+-- https://github.com/ucsd-progsys/liquidhaskell/pull/2540
+-- to fix
+-- https://github.com/ucsd-progsys/liquidhaskell/issues/1447
+
+module QuotRemInt where
+
+{-@ reflect intId @-}
+intId :: Int -> Int
+intId x = (x `quot` 2) * 2 + (x `rem` 2)
+
+{-@ reflect intId' @-}
+intId' :: Int -> Int
+intId' x = (x `div` 2) * 2 + (x `mod` 2)
+
+{-@ sumId :: x : Int -> y : Int -> {z : Int | z = x + y}@-}
+sumId :: Int -> Int -> Int
+sumId x y =  intId x + intId y
+
+{-@ sumId' :: x : Int -> y : Int -> {z : Int | z = x + y}@-}
+sumId' :: Int -> Int -> Int
+sumId' x y =  intId' x + intId' y
```

Comment:
These could be tested more directly with
```
{-@ lemmaQuotRem :: x:Int -> { intId x == x } @-}
lemmaQuotRem :: Int -> ()
lemmaQuotRem _ _ = ()

{-@ lemmaDivMod :: x:Int -> { intId' x == x } @-}
lemmaDivMod :: Int -> ()
lemmaDivMod _ _ = ()
```

Other tests to include here:
```
{-@
examplesQuot
   :: { -2 = quot -5 2
        && -2 = quot 5 -2
        && 2 = quot -5 -2
        && 2 = quot 5 2
        && 0 = quot 0 2
      }
@-}
examplesQuot :: ()
examplesQuot = ()
```
and similarly for `rem`.